tests/integration/targets/ec2_asg_instance_refresh/tasks/main.yml

```diff
@@ -0,0 +1,506 @@
+          created: "{{ resource_prefix }}-route"
+        routes:
+          - dest: 0.0.0.0/0
+            gateway_id: "{{ igw.gateway_id }}"
+        subnets:
+          - "{{ testing_subnet.subnet.id }}"
+
+    - name: create a security group with the vpc created in the ec2_setup
+      ec2_group:
+        name: "{{ resource_prefix }}-sg"
+        description: a security group for ansible tests
+        vpc_id: "{{ testing_vpc.vpc.id }}"
+        rules:
+          - proto: tcp
+            from_port: 22
+            to_port: 22
+            cidr_ip: 0.0.0.0/0
+          - proto: tcp
+            from_port: 80
+            to_port: 80
+            cidr_ip: 0.0.0.0/0
+      register: sg
+
+    - name: ensure launch configs exist
+      ec2_lc:
+        name: "{{ item }}"
+        assign_public_ip: true
+        image_id: "{{ ec2_ami_image }}"
+        user_data: |
+          package_upgrade: true
+          package_update: true
+          packages:
+            - httpd
+          runcmd:
+            - "service httpd start"
+        security_groups: "{{ sg.group_id }}"
+        instance_type: t3.micro
+      loop:
+        - "{{ resource_prefix }}-lc"
+        - "{{ resource_prefix }}-lc-2"
+
+    - name: launch asg and do not wait for instances to be deemed healthy (no ELB)
+      ec2_asg:
+        name: "{{ resource_prefix }}-asg"
+        launch_config_name: "{{ resource_prefix }}-lc"
+        desired_capacity: 1
+        min_size: 1
+        max_size: 1
+        vpc_zone_identifier: "{{ testing_subnet.subnet.id }}"
+        wait_for_instances: no
+        state: present
+      register: output
+
+    - assert:
+        that:
+        - "output.viable_instances == 0"
+
+  # ============================================================
+
+    - name: test invalid cancelation - V1 - (pre-refresh)
+      ec2_asg_instance_refresh:
+        name: "{{ resource_prefix }}-asg"
+        state: "cancelled"
+      ignore_errors: yes
+      register: result
+
+    - assert:
+        that:
+        - "'An error occurred (ActiveInstanceRefreshNotFound) when calling the CancelInstanceRefresh operation: No in progress or pending Instance Refresh found for Auto Scaling group {{ resource_prefix }}-asg' in result.msg"
+
+    - name: test starting a refresh with a valid ASG name - check_mode
+      ec2_asg_instance_refresh:
+        name: "{{ resource_prefix }}-asg"
+        state: "started"
+      check_mode: true
+      register: output
+
+    - assert:
+        that:
+        - output is not failed
+        - output is changed
+        - '"autoscaling:StartInstanceRefresh" not in output.resource_actions'
+
+    - name: test starting a refresh with a valid ASG name
+      ec2_asg_instance_refresh:
+        name: "{{ resource_prefix }}-asg"
+        state: "started"
+      register: output
+
+    - assert:
+        that:
+        - "'instance_refresh_id' in output.instance_refreshes"
+
+    - name: test starting a refresh with a valid ASG name - Idempotent
```

Comment:
can you add idepotence check_mode tests for this module? not sure if its possible given the module

Reply:
Added.